Graylog2/graylog2-server
# Remove entries with null value from `MapLiteralExpression` to prevent NPEs

## Description
Maps in pipeline rules can be expressed as `{key : "value"}`
When typing the rule source, NullPointerExceptions are thrown while the value isn't defined yet, e.g.
``` set_fields({key}) ```

Additionally, that had the effect that no error was shown in the frontend and the rule could not be saved, also with no error message being shown. 

This change makes sure that only complete key/value sets are used in `MapLiteralExpression`

## Motivation and Context
also fixes #14818 

## How Has This Been Tested?
- type in incomplete maps in the rule source editor
- no exceptions should be thrown 
- rule parsing errors should be shown in the editor
- use a valid map, e.g. in set_fields
- make sure all "complete" key/value pairs are present

## Types of changes
